frontend-react/src/ScrollToTop.tsx

```diff
@@ -0,0 +1,18 @@
+import { useEffect } from "react";
+import { useLocation } from "react-router-dom";
+
+export const ScrollToTop = () => {
+    const { pathname } = useLocation();
+
+    useEffect(() => {
+        const canControlScrollRestoration =
+            "scrollRestoration" in window.history;
+        if (canControlScrollRestoration) {
+            window.history.scrollRestoration = "manual";
+        }
+
+        window.scrollTo(0, 0);
+    }, [pathname]);
+
+    return null;
```

Comment:
Hmmmm!!!  I guess since we're not actually rendering anything and we're already using a `useEffect`, we could just effectively make this a custom hook that's invoked in the App component instead of needing to "render" a component 🤔

Reply:
This is true. The SO version that I copied returned children but at the same time said not to use it as a parent due to re-renders which is why I null'd it. As long as the hook's returned result never changes (thus never causing a re-render in the invoker) it would make a cleaner implementation for sure.